lib/collection/Cargo.toml

```diff
@@ -12,9 +12,10 @@ edition = "2021"
 tracing = ["dep:tracing", "api/tracing", "segment/tracing"]
 
 [dev-dependencies]
+async-std = { version = "1.12", features = ["attributes"] }
```

Comment:
Why though? 😳

Reply:
`rstest` requires it when a test function is `async`: https://github.com/la10736/rstest?tab=readme-ov-file#async

Reply:
Should we maybe use [this][rstest-inject-test-attr] with `#[tokio::test]` to not depend on whole `async-std`?

[rstest-inject-test-attr]: https://github.com/la10736/rstest?tab=readme-ov-file#inject-test-attribute

Reply:
Yeah...

We use `tokio::test` in other cases, but we can't do that here.